### PR TITLE
build: Clean up post-build copy of Windows DLLs

### PR DESCRIPTION
**Pull request by mrobinson**
- No longer use vcvarsall.bat at all. Instead find the Windows SDK
  directory by looking in the registry.
- Split logic for copying Windows dependencies into its own function and
  do some minor clean up, such as collecting all MSVC functionality into
  visual_studio.py.
- Remove support for Visual Studio 2015 and Visual Studio 2017.

This is a preparatory change in order to support Visual Studio 2022.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they are a build adjustment.

### CODE REVIEW

**Comment by mrobinson** [comment]
This needs a little work. I'll address the failures tomorrow.

**Comment by mrobinson** [comment]
Okay. This is ready to review!

File: python/servo/visual_studio.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+
+def find_windows_sdk_installation_path(vs_platform: str) -> str:
+    # This module must be imported here, because other platforms also
+    # load this file and the module is platform-specific.
+    import winreg
+
+    """Try to find the Windows SDK installation path using the Windows registry.
+    Raises and Exception if the path cannot be found in the registry."""
```
Comment (mukilan) [inline python/servo/visual_studio.py:87]: `Raises an exception`?

File: python/servo/visual_studio.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+def find_highest_msvc_version():
+    editions = ["Enterprise", "Professional", "Community", "BuildTools"]
+    prog_files = os.environ.get("ProgramFiles(x86)")
+    base_vs_path = os.path.join(prog_files, "Microsoft Visual Studio")
+
+    # TODO(mrobinson): Add support for Visual Studio 2022.
+    vs_versions = ["2019"]
+    versions = {
+        ("2019", "vs"): "16.0",
```
Comment (mukilan) [inline python/servo/visual_studio.py:46]: Is the second component needed in the key tuple? It seems be hardcoded in line 51?

Comment (mrobinson) [reply python/servo/visual_studio.py:46]: I don't think this is necessary. This was probably left over from some dead code. I'm happy to remove it.

File: python/servo/build_commands.py [hunk 3]
```diff
@@ -423,7 +362,36 @@ def package_gstreamer_dylibs(cross_compilation_target, servo_bin):
...
+        except StopIteration:
+            print(f"WARNING: could not find {dll_name}")
+
+    print(" • Copying ANGLE DLLs to binary directory...")
+    find_and_copy_built_dll("libEGL.dll")
+    find_and_copy_built_dll("libGLESv2.dll")
+
+    print(" • Copying GStreamer DLLs to binary directory...")
+    if not package_gstreamer_dlls(servo_exe_dir, target_triple):
+        return False
```
Comment (mukilan) [inline python/servo/build_commands.py:387]: We return False here, but the caller doesn't seem to check this return value. Also the type of this function doesn't specify that we are returning a boolean.

File: python/servo/visual_studio.py (hunk 1 above)
Comment (mrobinson) [reply python/servo/visual_studio.py]: Thanks! I've fixed this.

File: python/servo/visual_studio.py (hunk 2 above)
Comment (mrobinson) [reply python/servo/visual_studio.py]: I've removed this as well as cleaned up a bit some of the surrounding code, returning a `dataclass` from the caller, which should help ensure type safety.

File: python/servo/build_commands.py (hunk 3 above)
Comment (mrobinson) [reply python/servo/build_commands.py:388]: I've marked this method as returning a boolean and properly handle the result in the caller now.

**Review by mukilan — APPROVED** [review]

**Pull request merged**